python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -334,13 +334,18 @@ def _siblingRef(original, datasetType, dataId):
     Parameters
     ----------
     original : `lsst.daf.persistence.ButlerDataRef`
-        A dataRef related to the desired one.
+        A dataRef related to the desired one. Assumed to represent a unique dataset.
     datasetType : `str`
         The desired type of the new dataRef. Must be compatible
         with ``original``.
     dataId : `dict` from `str` to any
         A possibly partial data ID for the new dataRef. Any properties left
         unspecified shall be copied from ``original``.
     """
+    # Work around mismatched HDU lists for raw and processed data
+    cleanId = original.dataId.copy()
+    if 'hdu' in cleanId:
+        del cleanId['hdu']
```

Comment:
This method shouldn't be necessary: `dataId` munging like this shouldn't be needed at all . How does processCcd manage the case where 'hdu' is in the dataId?

Reply:
It doesn't do anything special (including when writing calexps).

As for the method as a whole being unnecessary, I think you're right. The original purpose (creating datarefs for calexp templates) would be better done using `Butler.subset` and a `run`-local butler.

Reply:
@parejkoj Looking again at the code, I think I originally introduced `_siblingRef` as an optimization for the following common case:

* The user specifies a (raw) data ID on the command line that restricts both visit and ccd, and
* The user specifies only a visit for `--templateId`, as [specified by `ImageDifferenceTask`](https://github.com/lsst/pipe_tasks/blob/master/python/lsst/pipe/tasks/imageDifference.py#L789-L790).

In fact, `GetCalexpAsTemplateTask` does exactly the same thing as `_siblingRef`: it [takes the primary dataRef and overwrites only its visit field](https://github.com/lsst/ip_diffim/blob/master/python/lsst/ip/diffim/getTemplate.py#L197-L198).

Anyway, I can replace the calls to `_siblingRef` with `butler.subset`, but this will mean that `ap_pipe` will always reduce an entire visit for `--templateId` even when only a few CCDs are needed.

Reply:
Oof. This sort of dataId munging is probably going to break with Gen3 butler. I guess we can leave it as is and deal with it when the new API is available.

Please file a ticket about how to handle the "dataRef and templateId" question with gen3 butler (and note the similar uses, e.g. `GetCalexpAsTemplateTask`), and reference it in `_siblingRef`'s docstring (probably in `Notes`?). I think we need much better guidance on this sort of thing, but getting that guidance is at some point in the future.

Reply:
I'm pretty sure *all* butler usage will break in version 3.

Where exactly do you expect me to file a ticket? For which component, to do what?